### PR TITLE
[stable/influxdb] update to support k8s v1.16

### PR DESCRIPTION
**Pull request by billimek**
#### What this PR does / why we need it:
Handle [API deprecations](https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/) in kubernetes v1.16

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by Aisuko** [comment]
@billimek Thanks for your PR, What do you think the content in this PR #17532? I mean the author adds lots of complexes hard code in the `_helpers.tpl` may not a good idea. But, Is there right that we need to make the tips to tell the end-user must be careful to upgrade that chart.

**Comment by billimek** [comment]
> @billimek Thanks for your PR, What do you think the content in this PR #17532? I mean the author adds lots of complexes hard code in the `_helpers.tpl` may not a good idea. But, Is there right that we need to make the tips to tell the end-user must be careful to upgrade that chart.

Hi, it's a good question.  The PR referenced makes a check for kubernetes version >=1.14-0 as a precondition to use the "apps/v1" API Version but apps/v1 has been available since kubernetes release v1.9.  It also seems to be necessary to add the necessary `matchLabels` to the spec in order for the apps/v1 to be consumed correctly.  This would need to also happen in PR #17532.

Additionally, it appears that the switch to apps/v1 (which is required in order for the chart to be able to deploy to kubernetes v1.16) may be a breaking change.  See [this issue](https://github.com/helm/helm/issues/6583) for some context.

Therefore, it may be necessary to bump the chart version to 1.6.0 or even 2.0.0 to denote the change.  What do you think?

File: stable/influxdb/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: influxdb
-version: 1.5.0
+version: 1.5.1
```
Comment (Aisuko) [inline stable/influxdb/Chart.yaml]: As we discussed, plase bump the major version.

**Review by Aisuko — CHANGES_REQUESTED** [review]
Thanks for your PR, please make the tips in the README.md and bump the chart major version.

**Comment by Aisuko** [comment]
#17343

**Comment by Aisuko** [comment]
#17343

File: stable/influxdb/Chart.yaml (hunk 1 above)
Comment (billimek) [reply stable/influxdb/Chart.yaml]: Done!

File: stable/influxdb/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: influxdb
-version: 1.5.0
-appVersion: 1.7.6
+version: 1.5.1
```
Comment (Aisuko) [inline stable/influxdb/Chart.yaml]: Please bump the chart major version to 2, `version: 2.0.0`

**Review by Aisuko — CHANGES_REQUESTED** [review]

File: stable/influxdb/Chart.yaml (hunk 2 above)
Comment (billimek) [reply stable/influxdb/Chart.yaml]: Apologies, I bumped the appVersion instead.  Fixed now.

**Comment by Aisuko** [comment]
Thanks a lot sir

**Pull request merged**